From: Luca Boccassi <luca.boccassi@microsoft.com>
Date: Fri, 26 Feb 2021 10:25:31 +0000
Subject: [PATCH] udev rules: add rule to create /dev/ptp_hyperv

From 32e868f058da8b90add00b2958c516241c532b70 Mon Sep 17 00:00:00 2001
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1917458
Origin: upstream, https://github.com/systemd/systemd/commit/32e868f058da8b90add00b2958c516241c532b70

As for the KVM case, necessary for network cards with
PTP devices when running a guest on HyperV
---
 rules.d/50-udev-default.rules.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rules.d/50-udev-default.rules.in b/rules.d/50-udev-default.rules.in
index cef78f9..ffa4b7a 100644
--- a/rules.d/50-udev-default.rules.in
+++ b/rules.d/50-udev-default.rules.in
@@ -89,4 +89,6 @@ KERNEL=="udmabuf", GROUP="kvm"
 
 SUBSYSTEM=="ptp", ATTR{clock_name}=="KVM virtual PTP", SYMLINK += "ptp_kvm"
 
+SUBSYSTEM=="ptp", ATTR{clock_name}=="hyperv", SYMLINK += "ptp_hyperv"
+
 LABEL="default_end"
